rotkehlchen/db/dbhandler.py

```diff
@@ -3265,3 +3266,25 @@ def should_save_balances(self, cursor: 'DBCursor') -> bool:
         period = settings.balance_save_frequency * 60 * 60
         now = ts_now()
         return now - last_save > period
+
+    def get_ethereum_nodes(self) -> Dict[str, NodeName]:
+        """Get information about all the ethereum nodes in the database. If the
+        only_with_weight param is set to True then only the nodes with weight != 0 are
+        returned.
+        """
+        with self.conn.read_ctx() as cursor:
+            cursor.execute('SELECT name, address, owned FROM open_nodes;')
+            return {
+                entry[0]: NodeName.deserialize_from_db(
+                    node_name=entry[0],
+                    endpoint=entry[1],
+                    owned=bool(entry[2]),
+                ) for entry in cursor
+            }
+
+    def update_ethereum_node_list(self, nodes: List[NodeName]) -> None:
+        with self.conn.write_ctx() as cursor:
```

Comment:
No here you use `db.user_write()`. Please read the code to see how it's done literally everywhere else.

```suggestion
        with self.user_write() as cursor:
```

Also keep in mind that if the function is used anywhere you may also be able to reuse cursors then you should pass it the cursor from outside.

This one at least for now does not seem to.